### PR TITLE
refactor: merge user config

### PR DESCRIPTION
**Pull request by Timeless0911**
## Summary

do not need to `mergeRsbuildConfig` twice

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: packages/core/tests/__snapshots__/config.test.ts.snap [hunk 1]
```diff
@@ -7,6 +7,7 @@ exports[`Should compose create Rsbuild config correctly > Merge Rsbuild config 1
       "dev": {
         "progressBar": false,
       },
+      "format": "esm",
```
Comment (fi3ework) [inline packages/core/tests/__snapshots__/config.test.ts.snap:10]: We're adding unknown field to Rsbuild config here, could we prevent this?

**Review by fi3ework — APPROVED** [review]

File: packages/core/tests/__snapshots__/config.test.ts.snap (hunk 1 above)
Comment (Timeless0911) [reply packages/core/tests/__snapshots__/config.test.ts.snap:10]: Just add format this time, and other unknown field is already there for severral days. Rsbuild inspect seems to filter them in display.

Comment (fi3ework) [reply packages/core/tests/__snapshots__/config.test.ts.snap:10]: > and other unknown field is already there for severral days

Did we? The seems not in the snapshot at the moment.

Comment (Timeless0911) [reply packages/core/tests/__snapshots__/config.test.ts.snap:10]: because we only add format in our unit test

**Pull request merged**